packages/assets-controllers/src/CurrencyRateController.ts

```diff
@@ -202,30 +196,44 @@ export class CurrencyRateController extends BaseControllerV2<
   }
 
   /**
-   * Updates exchange rate for the current currency.
+   * Updates exchange rates for the network.
    *
+   * @param newCurrentCurrency - An optional new ISO 4217 currency code
+   * to switch to. If omitted, the existing currency code will be used.
+   * @param newNativeCurrency - An optional new symbol for the network's
+   * base asset. If omitted, the existing native currency will be used.
    * @returns The controller state.
    */
-  async updateExchangeRate(): Promise<CurrencyRateState | void> {
+  async updateExchangeRate(
+    newCurrentCurrency?: string,
+    newNativeCurrency?: string,
+  ): Promise<CurrencyRateState | void> {
     if (!this.#enabled) {
       console.info(
         '[CurrencyRateController] Not updating exchange rate since network requests have been disabled',
       );
       return this.state;
     }
+
     const releaseLock = await this.mutex.acquire();
-    const {
-      currentCurrency: stateCurrentCurrency,
-      nativeCurrency: stateNativeCurrency,
-      pendingCurrentCurrency,
-      pendingNativeCurrency,
-    } = this.state;
+
+    if (newCurrentCurrency !== undefined) {
+      this.update((state) => {
+        state.pendingCurrentCurrency = newCurrentCurrency;
+      });
+    }
+
+    if (newNativeCurrency !== undefined) {
+      this.update((state) => {
+        state.pendingNativeCurrency = newNativeCurrency;
+      });
+    }
```

Comment:
The pending state looks less required now that the values are available as function arguments.  But the pending state may be needed to fix a related bug.  When switching networks, MM extension temporarily uses the conversion rate from the old network until the cryptocompare calls return for the new network. (More visible on slow internet):

https://github.com/MetaMask/core/assets/3500406/71027e66-e19c-4511-9124-97a745a26951

Balances are usually available before conversion rates, so I think clients should be consulting the pending state to know they can't show fiat values until rates are available for the new network.